### PR TITLE
Integrate libsbp

### PR DESCRIPTION
**Pull request by mfine**
Integrate libsbp into the firmware:

1. Add libsbp submodule
2. Update Makefile machinery to build and link a static libsbp library
3. Bring in libswiftnav's sbp_utils to the firmware
4. Update from libswiftnav to libsbp based paths where applicable

The submodule branches supporting this request are:

https://github.com/swift-nav/libsbp/pull/40 -> [libsbp/mfine-remove-crc24q](https://github.com/swift-nav/libsbp/compare/mfine-remove-crc24q) 

https://github.com/swift-nav/libswiftnav/pull/96 -> [libswiftnav/mfine-remove-sbp](https://github.com/swift-nav/libswiftnav/compare/mfine-remove-sbp)

These changes should get some good time on the test bed!

/cc @mookerji @fnoble

### CODE REVIEW

**Comment by mfine** [comment]
retest this please

**Comment by mookerji** [comment]
add to whitelist

**Comment by mookerji** [comment]
retest this please.

**Comment by fnoble** [comment]
This is fantastic, so there weren't any changes to the structs that required changes to the firmware?

I guess lots of the code is still using `sbp_piksi.h` so I suppose the next step is to migrate that stuff.

**Comment by mfine** [comment]
Running overnight in the continuous builds. Looking good so far (not sure how to evaluate the quality yet :(). Will look into killing off `src/sbp_piksi.[ch]` and move on to what's in libsbp.

**Comment by mfine** [comment]
Overnight tests look good. Running locally on a piksi and connecting to it through the console looks good. Going to rebase things and get ready for a merge.

File: src/sbp.h [hunk 1]
```diff
@@ -13,9 +13,9 @@
 #ifndef SWIFTNAV_SBP_H
 #define SWIFTNAV_SBP_H
 
-#include <libswiftnav/common.h>
-#include <libswiftnav/sbp.h>
-#include <libswiftnav/sbp_messages.h>
+#include <libsbp/common.h>
+#include <libsbp/sbp.h>
+#include <libsbp/sbp_messages.h>
```
Comment (mookerji) [inline src/sbp.h]: .... so `libsbp/sbp_messages.h` is actually the legacy header moved from libswiftnav to libsbp containing the messages that are now split up across `libsbp/standard.h` and `libsbp/navigation.h`. `libsbp/sbp_messages.h` needs to be removed from from the libsbp c includes.

Comment (mfine) [inline src/sbp.h]: Ahh, good - nice. Will remove.

File: src/sbp_utils.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SWIFTNAV_SBP_UTILS_H
+#define SWIFTNAV_SBP_UTILS_H
+
+#include <libsbp/common.h>
+#include <libsbp/sbp_messages.h>
```
Comment (mookerji) [inline src/sbp_utils.h]: Similarly, libsbp/sbp_messages.h needs to be removed.

**Comment by mfine** [comment]
Rebased.

**Comment by mfine** [comment]
Incorporated in #281.

**Pull request closed without merging**